src/app/services/interceptor/api.interceptor.ts

```diff
@@ -3,24 +3,33 @@ import { HttpEvent, HttpHandler, HttpInterceptor, HttpRequest, HttpErrorResponse
 import { Observable, throwError } from 'rxjs';
 import { catchError } from 'rxjs/operators';
 import { Router } from '@angular/router';
-import { environment } from 'src/environments/environment';
 @Injectable({
   providedIn: 'root',
 })
 export class ApiInterceptor implements HttpInterceptor {
-  constructor(private router: Router) {}
+  constructor(private router: Router) { }
   intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
     const token = localStorage.getItem('accToken');
     let authReq = req;
 
     if (token) {
-      if (req.url.includes('/logout')) {
+      if (req.url.includes('/logout') || req.url.includes('/user/update') ||
```

Comment:
see if this statement can be outsourced to a new function its hard to read now

Reply:
and so is the whole function break it into seperate functions wherever possible and necessary